source/cipher_openssl.c

```diff
@@ -140,17 +141,27 @@ size_t aws_cryptosdk_md_size(enum aws_cryptosdk_md_alg md_alg) {
 }
 
 int aws_cryptosdk_md_update(struct aws_cryptosdk_md_context *md_context, const void *buf, size_t length) {
+    AWS_PRECONDITION(aws_cryptosdk_md_context_is_valid(md_context));
+    AWS_PRECONDITION(AWS_MEM_IS_READABLE(buf, length));
```

Comment:
is `AWS_MEM_IS_READABLE`updated?

Reply:
This is ok, because it will only be used in assert context.